.flake8

```diff
@@ -0,0 +1,13 @@
+[flake8]
+max-line-length = 99
+select: E,W,F,C,N
+exclude:
+  venv
+  .git
+  build
+  dist
+  lib/charms
+  *.egg_info
+ignore:
+  E501
```

Comment:
What is the reason for these ignores please?

Reply:
E501 is "line too long", but splitting the affected lines is causing `black` to fail
```
-            "https://login.ubuntu.com/+saml":
-                "32:15:20:9F:A4:3C:8E:3E:8E:47:72:62:9A:86:8D:0E:E6:CF:45:D5"
+            "https://login.ubuntu.com/+saml": "32:15:20:9F:A4:3C:8E:3E:8E:47:72:62:9A:86:8D:0E:E6:CF:45:D5"
```
```
"command":
-                        f"sh -c '{SCRIPT_PATH}/pod_start >>/srv/discourse/discourse.log 2&>1'",
+                    "command": f"sh -c '{SCRIPT_PATH}/pod_start >>/srv/discourse/discourse.log 2&>1'",

```
W503 is "line break before binary operator". This has been ignored as it is the recommended way to split lines as in 
```
self.config.get("s3_enabled")
            and (
                s3info.enabled != self.config.get("s3_enabled")
                or s3info.region != self.config.get("s3_region")
                or s3info.bucket != self.config.get("s3_bucket")
                or s3info.endpoint != self.config.get("s3_endpoint")
            )
```

Reply:
Ah, do you think it is worth adding an explanatory comment above the ignores so that others know these reasons in the future?

Reply:
I think the explanation and it'll be difficult to provide more context than what the rule is about